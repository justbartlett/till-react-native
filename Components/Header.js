import React from 'react';
import UserInfo from '../Components/UserInfo';
import {Colors} from '../Theme/Colors';
import LinearGradient from 'react-native-linear-gradient';
import RectangleSvg from '../img/rectangle.svg';

import {StyleSheet, View, Text} from 'react-native';

const Header = () => {
  return (
    <>
      <View style={styles.linearGradientMask}>
        <LinearGradient
          start={{x: 1.0, y: 1.0}}
          end={{x: 0.0, y: 0.0}}
          colors={[Colors.peachy, Colors.purpley]}
          style={styles.linearGradient}
        />
        <RectangleSvg style={styles.firstRectangle} width={124} height={132} />
        <RectangleSvg style={styles.secondRectangle} width={124} height={132} />
        <RectangleSvg style={styles.thirdRectangle} width={124} height={132} />
      </View>
      <View style={styles.layoutOffset} />
      <UserInfo />
      <View style={styles.transactions}>
        <Text style={styles.heading}>Transactions</Text>
      </View>
    </>
  );
};

export default Header;

const styles = StyleSheet.create({
  transactions: {height: 20},
  firstRectangle: {
    position: 'absolute',
    top: -50,
    left: -20,
  },
  secondRectangle: {
    position: 'absolute',
    top: 15,
    left: '35%',
  },
  thirdRectangle: {
    position: 'absolute',
    top: -20,
    right: -40,
  },
  heading: {
    color: Colors.white,
    fontSize: 16,
    marginLeft: 35,
    height: 20,
    flex: 1,
    fontFamily: 'NunitoSans-ExtraBold',
  },
  layoutOffset: {margin: -80},
  linearGradient: {
    height: 200,
  },
  linearGradientMask: {
    borderBottomLeftRadius: 40,
    borderBottomRightRadius: 40,
    overflow: 'hidden',
  },
});
