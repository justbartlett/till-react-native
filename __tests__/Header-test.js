/**
 * @format
 */

import 'react-native';
import React from 'react';
import Header from '../Components/Header';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
