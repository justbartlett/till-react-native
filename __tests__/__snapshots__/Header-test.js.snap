// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "borderBottomLeftRadius": 40,
        "borderBottomRightRadius": 40,
        "overflow": "hidden",
      }
    }
  >
    <BVLinearGradient
      colors={
        Array [
          4294949839,
          4289095121,
        ]
      }
      endPoint={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      locations={null}
      startPoint={
        Object {
          "x": 1,
          "y": 1,
        }
      }
      style={
        Object {
          "height": 200,
        }
      }
    />
    <SvgMock
      height={132}
      style={
        Object {
          "left": -20,
          "position": "absolute",
          "top": -50,
        }
      }
      width={124}
    />
    <SvgMock
      height={132}
      style={
        Object {
          "left": "35%",
          "position": "absolute",
          "top": 15,
        }
      }
      width={124}
    />
    <SvgMock
      height={132}
      style={
        Object {
          "position": "absolute",
          "right": -40,
          "top": -20,
        }
      }
      width={124}
    />
  </View>,
  <View
    style={
      Object {
        "margin": -80,
      }
    }
  />,
  <View
    style={
      Object {
        "height": 80,
        "marginLeft": "10%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View>
        <Image
          source={
            Object {
              "uri": "http://www.fillmurray.com/200/200",
            }
          }
          style={
            Object {
              "borderColor": "#FFFFFF",
              "borderRadius": 100,
              "borderWidth": 4,
              "height": 52,
              "marginRight": 15,
              "width": 52,
            }
          }
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "NunitoSans-Black",
              "fontSize": 22,
              "marginBottom": 4,
            }
          }
        >
          Johnny Ray
        </Text>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "NunitoSans-ExtraBold",
              "fontSize": 13,
            }
          }
        >
          My Balance: $
          998913.61
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "height": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "flex": 1,
          "fontFamily": "NunitoSans-ExtraBold",
          "fontSize": 16,
          "height": 20,
          "marginLeft": 35,
        }
      }
    >
      Transactions
    </Text>
  </View>,
]
`;
